### PR TITLE
[#43] handle survey units listing

### PR DESCRIPTION
**Pull request by davdarras**
Handle survey unit listing
fix i18n

### CODE REVIEW

File: src/i18n-fr.js [hunk 1]
```diff
@@ -36,4 +36,8 @@ export const messagesFr = {
   questionnaire_delete_confirmation_agree: "Confirmer la suppression",
   questionnaire_delete_confirmation_disagree: "Annuler",
   questionnaire_delete_success: "Questionnaire supprimé",
+  survey_unit_list_label: "Unités enquêté",
```
Comment (SimonDmz) [inline src/i18n-fr.js:39]: enquêtées

File: src/ui/pages/SurveyUnitListPage.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  const notifier = useNotifier();
+  const orchestratorUrl = process.env.REACT_APP_ORCHESTRATOR_URL;
+  const { questionnaireId, modeName } = useParams<SurveyUnitParams>();
+
+  useEffect(() => {
+    // react router doesn't type, but a check has already been done in Application.tsx ... (\\d+))
+    if (!(questionnaireId && modeName)) {
+      return;
+    }
+    const questionnaireIdNumber = parseInt(questionnaireId);
```
Comment (MickaelMenet) [inline src/ui/pages/SurveyUnitListPage.tsx:44]: Use radix 10 for parseInt

File: src/ui/pages/SurveyUnitListPage.tsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        <Block>
+          <Loader isLoading={isSurveyUnitsLoading && isQuestionnaireLoading}>
+            <Title>
+              {intl.formatMessage({ id: "survey_unit_list_label" })}
+            </Title>
+
+            <Subtitle>
+              <>
+                {questionnaire?.label}
+                <br />
```
Comment (MickaelMenet) [inline src/ui/pages/SurveyUnitListPage.tsx:81]: The use of the br tag either in semantics or for visual readers is not to be used it seems to me. ( to validate)

File: src/ui/pages/SurveyUnitListPage.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                  <TableBody>
+                    {questionnaire &&
+                      surveyUnitsData?.surveyUnits?.map((surveyUnit) => (
+                        <TableRow key={surveyUnit.id}>
+                          <TableCell component="th" scope="row">
+                            {surveyUnit.displayableId}
+                          </TableCell>
+                          <TableCell align="center">
+                            <a
+                              target="_blank"
```
Comment (MickaelMenet) [inline src/ui/pages/SurveyUnitListPage.tsx:115]: the target blank is not very accessible or you have to put additional information.

File: yarn.lock [hunk 5]
```diff
@@ -1935,7 +1935,7 @@
   dependencies:
     "@babel/runtime" "^7.19.0"
 
-"@mui/lab@^5.0.0-alpha.115":
+"@mui/lab@5.0.0-alpha.115":
```
Comment (MickaelMenet) [inline yarn.lock:1938]: use of the alpha version ?

File: src/ui/components/QuestionnaireEditForm.tsx [hunk 6]
```diff
@@ -269,7 +274,9 @@ export const QuestionnaireEditForm = memo(
                 select
                 onChange={handleContextChange}
                 error={errorContextInput ? true : false}
-                value={questionnaire.context}
+                value={
+                  questionnaire.context.name ? questionnaire.context.name : ""
```
Comment (SimonDmz) [inline src/ui/components/QuestionnaireEditForm.tsx:278]: `value={ questionnaire.context.name ?? ""}`
ça évite d'évaluer deux fois questionnaire.context.name

File: src/ui/pages/SurveyUnitListPage.tsx [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    setSurveyUnitsLoading(true);
+    surveyUnitUseCase
+      .getSurveyUnitsData(questionnaireId, mode)
+      .then((surveyUnitsData) => {
+        setSurveyUnitsData(surveyUnitsData);
+        setSurveyUnitsLoading(false);
+      })
+      .catch((err) => {
+        notifier.error(intl.formatMessage({ id: "error_request_failed" }));
+        console.log(err);
```
Comment (SimonDmz) [inline src/ui/pages/SurveyUnitListPage.tsx:65]: error handling doesn't remove the Loader => expected behavior ?

File: src/ui/components/QuestionnaireEditForm.tsx (hunk 6 above)
Comment (davdarras) [reply src/ui/components/QuestionnaireEditForm.tsx:278]: didn't know the Nullish Coalescing operator, thanks :)

File: src/ui/pages/SurveyUnitListPage.tsx (hunk 3 above)
Comment (davdarras) [reply src/ui/pages/SurveyUnitListPage.tsx:81]: Not really sure about this one. I'm not considering this br used as presentation layout, it's a part of the title content

File: yarn.lock (hunk 5 above)
Comment (davdarras) [reply yarn.lock:1938]: It is a bit confusing. I'm only using LoadingButton component referenced in mui-core documentation (https://mui.com/material-ui/react-button/#loading-button)), with no details about using an alpha version on a third party lib (no stable one). Nonetheless i should have seen it earlier. I will investigate later

**Pull request merged**